airflow/models/taskinstance.py

```diff
@@ -1550,11 +1550,21 @@ def _run_finished_callback(
     if callbacks:
         callbacks = callbacks if isinstance(callbacks, list) else [callbacks]
         for callback in callbacks:
-            log.info("Executing %s callback", callback.__name__)
+            if hasattr(callback, "__name__"):
+                log.info("Executing %s callback", callback.__name__)
+            elif hasattr(callback, "__class__"):
+                log.info("Executing %s callback", callback.__class__.__name__)
+            else:
+                log.info("Executing %s callback", callback.__code__)
```

Comment:
Honestly, why do we not just do `log.info("Executing %s callback", callback)`?

Reply:
+1, I'm unsure what is the context here, seems @romsharon98 has tried it before[[link](https://github.com/apache/airflow/pull/38892#discussion_r1566141583)], do we expect the actual name of a callable instead of `repr` one

Reply:
yes callback itself provides the good enough information.
Perhaps we are much like to the "position" instead of actual name.

Take the following example

if we have on_success_callback = [
  slack1,
  slack2,
  slack3
]

with only "name" we can't distinguish which "slack" failed.

Reply:
Yeah position in list + the function object itself sounds like the best approach to me.